active/0000-routeable-components.md

```diff
@@ -0,0 +1,488 @@
+- Start Date: 2015-02-27
+- RFC PR:
+- Ember Issue:
+
+# Summary
+
+Eliminates Controllers.
+
+# Motivation
+
+As Ember has evolved toward a component-centric model, controllers
+have become redundant and vestigial. This unnecessarily burdens
+developers with learning an additional concept and forces them to make
+unnecessary choices.
+
+The goal of this design is to describe how to eliminate
+`Ember.Controller` while maintaining sufficiently rich APIs to do all
+the important things controllers do.
+
+# Scope
+
+- Eliminate `Controller` by designing alternative APIs for routing to
+  a component, maintaining long-lived non-URL state, and using query
+  parameters.
+
+- Simplify the behavior of `Route`'s model hooks. While this is not
+  strictly needed to achieve the goal of eliminating controllers, it
+  makes sense to address it while we're redesigning the API between
+  routes & components anyway.
+
+- Consider how routable components relate to asynchronous components,
+  because they may share concepts or implementation.
+
+- Clarify the distinction between Components and Fragments ("tagless
+  components").
+
+# Definitions
+
+ - **routable component** - a component whose identity and attributes
+   are determined from the URL at runtime. Routable components are
+   what go inside `{{outlet}}`s.
+
+ - **asynchronous component** - a component with built-in support for
+   asynchronously loading data, with a friendly API for reflecting
+   loading and errors states.
+
+Both of the above are conceptual definitions for the purposes of this
+design document. They are not necessarily actual classes or concepts
+that would need to be taught to new Ember devs.
+
+# Detailed design
+
+## Routing to a component
+
+Our current API for letting a `Route` decide which things to render
+and where they should go is fairly rich and complex. We can simplify
+it significantly.
+
+### Choosing what component(s) to render
+
+The current `Route.render` method is where a `Route` expresses what
+should actually get rendered. It has a fair bit of complexity that we
+can deprecate. Today you can say things like:
+
+````js
+this.render('post', {
+  view: 'special-post',
+  controller: 'other-controller' ,
+  model: myPost
+})
+````
+
+Which lets you pick and choose which template, view, controller, and
+model to render. There are fairly rich semantics for choosing defaults
+for each of those things, and allowing views to choose their own
+template, etc.
+
+Instead, we can deprecate the use of the `view`, `controller`, and
+`model` arguments, and accept a name which will be used to lookup a
+*component*, not a template. We will also accept arbitrary attributes
+to set on the component.
+
+````js
+this.render('post', attributes)
```

Comment:
From your [previous comment](https://github.com/emberjs/rfcs/pull/38#issuecomment-76535592) it sounds like the solution is just "they are still components....but don't follow the dash rule". That seems like it will be quite confusing for some, is my concern unfounded?